packages/sitecore-jss-cli/src/create/index.ts

```diff
@@ -3,73 +3,144 @@ import fs from 'fs';
 import glob from 'glob';
 import path from 'path';
 
+interface PackageJsonConfig {
+  appName: string;
+  graphQLEndpointPath?: string;
+  sitecoreDistPath?: string;
+}
+
+interface PackageJson {
+  name: string;
+  config: PackageJsonConfig;
+}
+
 /**
- * @param {string} projectFolder
- * @param {string} name
- * @param {string} hostName
+ * Escapes a string literal for use in RegEx
+ * @param {string} literal
  */
-export function applyNameToProject(projectFolder: string, name: string, hostName: string) {
-  console.log(chalk.cyan(`Setting package name ${name}...`));
+function escapeRegex(literal: string) {
+  return literal.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+}
 
-  const packagePath = path.join(projectFolder, 'package.json');
+/**
+ * @param {PackageJson} pkg package.json object
+ * @param {string} name App name
+ * @param {string} [replaceName] Optional token which will enable "name replacement mode". If omitted, default/conventional values are used.
+ */
+export function applyNameToPackageJson(pkg: PackageJson, name: string, replaceName?: string) {
+  pkg.name = name; // root name will never match "replaceName", so always set directly
+  pkg.config.appName = replaceName
+    ? pkg.config.appName.replace(RegExp(escapeRegex(replaceName), 'g'), name)
```

Comment:
It seems unusual to initiate a RegEx with something that just ran through a string replace, as RegEx can perform replacements on matches, and include those in its result. Is it possible to let RegEx handle the escaping?

Reply:
@anastasiya29 The escape is there to make sure any special regex characters in the replaceName token are treated as literals in the search (and need to specifically use RegExp as first param). Illia had a comment about this bit, and has been refactored. I think it's more clear in the latest update, specifically the [related test](https://github.com/Sitecore/jss/blob/5c37cc7415f428166b13a97685e73a8acf5de906/packages/sitecore-jss-cli/src/create/index.test.ts#L40).